hw_4/task_2.js

```diff
@@ -0,0 +1,72 @@
+//=============================================
+// Сравнение пицц с конкурентами
+//=============================================
+console.log('============= Сравнение пицц с конкурентами ====================')
+
+
+function pizzaValidator(ourDeliciousPizza) {
+    const result = [];
+    const theirTastelessPizza = ['Peperoni', 'Caprichosa', 'Diablo', '4 chesee', 'hawai']
+    ourDeliciousPizza.forEach(pizza => { if (!theirTastelessPizza.includes(pizza)) result.push(pizza) });
+    if (result.length == 0) return null
+    else return result;
+}
+
+//Debug
+//const ourPizza  = ['Peperoni','Caprichosa','Diablo','4 chesee','hawai']
+//const ourPizza  = ['Peperoni','Caprichosa','Diablo','4 chesee','hawai',"homemade Delicious"]
+//const ourPizza  = ['Peperoni','Caprichosa']
+//const ourPizza  = ['Diablo','Diablo','Diablo','Diablo','Diablo']
+const ourPizza = ['Peperoni', 'Caprichosa', 'Diablo', '4 chesee', 'hawai']
+
+console.log(pizzaValidator(ourPizza))
+ 
+//=============================================
+// Найти всамые длинные слова
+//=============================================
+console.log('============= Найти всамые длинные слова ====================')
+
+
+function parseSentence(sentence) {
+ 
+    const worlds = sentence.matchAll(/([А-Я,а-я,ё]+)/g) ;  //object
+    
+    const arr = Array.from(worlds); 
+    const maxLength =  Math.max(...arr.map(el => el[0].length));
```

Comment:
It's enough to use one loop to iterate and collect words
Use `return` key word for result value